### DISCUSSION TITLE
Clarify ubuntu minimum version of Ubuntu 20

### DISCUSSION THREAD

**Discussion by Ryanf55** [eProsima/Fast-RTPS-docs#899] [General]
## Description

Looks like Ubuntu 18 no longer works to build FastDDS from source due the following:

1. CMake is too old if you install it from apt
```
root@docker-desktop:~/Fast-DDS/foonathan_memory_vendor/build# cmake .. -DCMAKE_INSTALL_PREFIX=~/Fast-DDS/install -DBUILD_SHARED_LIBS=ON
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.14 or higher is required.  You are running version 3.10.2
  ```
  
 2. It requires the version of libasio on Ubuntu 22. While this works with FastDDS 2.6.6, it does not in 2.6.9. Although these are patch versions, there was an ABI break by introducing a dependency on a verion of asio only available on Ubuntu 20.
 
![image](https://github.com/user-attachments/assets/1dd76cc5-af81-4e00-84a3-317c4b5d3079)

If it is not the intent of eProsima to support Ubuntu 18, can you please verify a minimum requirement of Ubuntu 20 in the source install page?

## Demo

```bash
docker run --net host -it ubuntu:18.04 bash
apt update
apt install -y libasio-dev libtinyxml2-dev  libssl-dev softhsm2 git cmake
mkdir ~/Fast-DDS
cd ~/Fast-DDS
git clone https://github.com/eProsima/foonathan_memory_vendor.git
mkdir foonathan_memory_vendor/build
cd foonathan_memory_vendor/build
cmake .. -DCMAKE_INSTALL_PREFIX=~/Fast-DDS/install -DBUILD_SHARED_LIBS=ON
```

**Comment by JesusPoderoso** [comment]
Hi @Ryanf55, thanks for using Fast DDS.
Unfortunately, Fast DDS 2.6.x (from v2.6.0 to latest v2.6.9) is not officially supported for Ubuntu 18 (refer to [platform support doc](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md) for [2.6.x](https://fast-dds.docs.eprosima.com/en/latest/notes/versions.html#build-system-dependencies), [v2.6.0](https://github.com/eProsima/Fast-DDS/blob/v2.6.0/PLATFORM_SUPPORT.md#platforms), and [v2.6.9](https://github.com/eProsima/Fast-DDS/blob/v2.6.9/PLATFORM_SUPPORT.md#platforms) for more information).
You may consider contacting [commercials support](https://github.com/eProsima/Fast-DDS/tree/master#getting-help) for premium support options.
I am moving this ticket to the discussions section according to the [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md).

**Comment by MiguelCompany** [comment]
I was writing an answer to this but @JesusPoderoso was quicker.

The root folder of the repository has a `PLATFORM_SUPPORT.md` that indicates the supported platforms.
On v2.6.0 that file (see [here](https://github.com/eProsima/Fast-DDS/blob/v2.6.0/PLATFORM_SUPPORT.md)) indicates `Ubuntu Focal (20.04)`

In the documentation we have [this section](https://fast-dds.docs.eprosima.com/en/stable/notes/versions.html#build-system-dependencies) that for 2.6.x tells that CMake 3.16 is required, and Ubuntu Focal (20.04) with GCC 9 is the Tier 1 platform.

I suppose it would be nice to have a link to that section in the build from sources one.

**Reply by Ryanf55** [reply]
Thanks for the info, yep adding a reminder link would be fantastic to the beginning of the page.
